### DISCUSSION TITLE
How to improve SpamAssassin heuristics?

### DISCUSSION THREAD

**Discussion by MarcS1975** [docker-mailserver/docker-mailserver#4040] [Q&A]
I am having trouble getting legitimate mail through Spamassesin. Even emails from domains like letsencrypt.com are marked as spam.
In your documentation I found a few good [tips ](https://docker-mailserver.github.io/docker-mailserver/v12.0/faq/#how-can-i-manage-my-custom-spamassassin-rules)about enhancing the rules SA is using to detect spam but the real problem is often false positive HAM.
Do you have any tips around SA rules to improve Ham detection?

**Comment by glime-ai** [comment]
**Please note** that this response was created using generative AI, leveraging previously available information from the repository. This information may be helpful. However, it is our strong recommendation that you carefully consider the information before blindly applying it.

**This response is generated only once - you will not be able to interact with the glime-ai.**

---

To improve the detection of legitimate emails (HAM) and reduce false positives in SpamAssassin, you can follow these tips:

1. **Train SpamAssassin with HAM and SPAM**: Ensure that SpamAssassin is trained with enough examples of both legitimate emails (HAM) and spam. SpamAssassin requires training with at least 200 emails each of HAM and SPAM to start effectively filtering. You can automate this training using cron jobs as described in the documentation.

2. **Adjust Scoring Rules**: Customize the scoring rules in SpamAssassin to better suit your email patterns. You can decrease the scores of rules that are frequently flagging legitimate emails as spam. This involves editing the `local.cf` file or creating custom rule files.

3. **Whitelist Trusted Senders**: Add trusted sender domains or email addresses to a whitelist. This can be done by modifying the `whitelist_from` directive in your SpamAssassin configuration.

4. **Use Auto-Whitelisting**: Enable auto-whitelisting in SpamAssassin, which automatically adjusts scores for senders based on their history of sending non-spam emails.

5. **Check Network Tests**: Ensure that DNS-based tests (like SPF, DKIM, and DMARC) are correctly set up and not contributing to false positives. These tests can sometimes incorrectly flag legitimate emails as spam if not configured properly.

6. **Review and Adjust Headers**: Check if certain headers in your legitimate emails are causing them to be scored as spam and adjust the rules accordingly.

For detailed guidance on configuring these settings, you can refer to the SpamAssassin documentation and the specific sections on managing custom rules and configurations in the Docker Mailserver documentation.

This comment was generated by [Glime](https://glimelab.ai/?utm_source=0d5b86fb7eea4a54acac405212c29aba&utm_medium=github).

**Comment by casperklein** [comment]
I use spamassassin with no adjustments and never encountered such problems. Do you use DNS block lists?

**Comment by MarcS1975** [comment]
no DNS blocker but using KAM=1

**Comment by MarcS1975** [comment]
I think something is not right. Since today emails from Bloomberg (which I
have been subscribed to for 3 months) suddenly go into the spam folder.

**Reply by casperklein** [reply]
Did you try disabling KAM?
